### DISCUSSION TITLE
Improved file uploader

### DISCUSSION THREAD

**Discussion by alexmanase** [livewire/flux#894] [Feature Requests] [answered]
A File Uploader, where you can preview the image, something like a lighter Filepond.

This would be nice to include customisation for:

- adding/replacing a file
- removing the current file

For example this "Image preview" in Mary UI: https://mary-ui.com/docs/components/file#image-preview

Or this part from Jetstream: https://github.com/laravel/jetstream/blob/5.x/stubs/livewire/resources/views/profile/update-profile-information-form.blade.php#L13-L52

**Comment by yamcomnet** [comment]
Laravel Nova also has quite nice file field. Simple and functional.
https://nova.laravel.com/docs/v5/resources/file-fields

**Comment by amshehzad** [comment]
+1
Really looking forward to it!

**Comment by NimaKhadivar** [comment]
any news

**Comment by jeffchown** [comment]
Nothing Flux-specific yet.  I've found these helpful:
- https://codecourse.com/watch/build-a-livewire-drag-drop-uploader/introduction-and-demo-livewire-drag-and-drop-uploader
- https://codecourse.com/courses/chunking-large-uploads-in-livewire
- https://codecourse.com/courses/crop-and-upload-profile-photos-with-livewire

(disclaimer: I'm not affiliated with CodeCourse)

**Comment by joshhanley** [comment, marked as answer]
We've just added support for a dedicated file upload in v2.5!

You can find the documentation here https://fluxui.dev/components/file-upload
